lib/index.d.ts

```diff
@@ -0,0 +1,43 @@
+import {Request} from 'express';
+import * as passport from 'passport';
+
+export interface StrategyOptions {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback?: true;
+  scope?: string[];
+}
+
+export interface StrategyOptionsWithRequest {
+  clientID: string;
+  clientSecret: string;
+  callbackURL: string;
+  passReqToCallback: true;
+  scope?: string[];
+}
+
+export interface VerifyOptions {
+  message: string;
+}
+
+export interface VerifyCallback {
+  (error: any, user?: any, options?: VerifyOptions): void;
+}
+
+export interface VerifyFunctionWithRequest {
+  (req: Request, accessToken: string, refreshToken: string, profile: any, done: VerifyCallback): void;
+}
+
+export interface VerifyFunction {
+  (accessToken: string, refreshToken: string, profile: any, done: VerifyCallback): void;
+}
+
+declare class Strategy implements Strategy {
+  public name: string;
+  public authenticate: (req: Request, options?: object) => void;
```

Comment:
It also has a public function called `authenticate` which accepts a `req` which again uses the express Request type, and an optional options object (there's a mouthful). This is not of type any, it's of type "object" so it can't be a primative. This is similar to the use of `any` but doesn't allow primatives. Again because this is mostly an internally used function I didn't want to get too strict about it to allow for changes in this module. Also because we just pass it in and never see it again, it's not going to bother us much when using the module, unlike the verify functions where we may end up adding al ot of business logic.